bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,234 @@
+            updateStatus(ThingStatus.ONLINE);
+            Runnable heartbeatpolling = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        if (YIOREMOTEHANDLESTATUS_actualstatus.equals(YIOREMOTEHANDLESTATUS.AUTHENTICATED)) {
+                            YIOremoteDockwebSocket_ClientSocket.sendMessage(YIOREMOTEMESSAGETYPE.HEARTBEAT, "");
+
+                            Thread.sleep(1000);
+
+                            if (YIOremoteDockwebSocket_ClientSocket.get_boolean_heartbeat()) {
+                                logger.debug("heartbeat ok");
+                                updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS,
+                                        YIOremoteDockwebSocket_ClientSocket.get_string_receivedstatus());
+                            } else {
+                                logger.warn("Connection lost no ping from YIO DOCK");
+                                YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.CONNECTION_FAILED;
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                        "Connection lost no ping from YIO DOCK");
+                                updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+                                pollingJob.cancel(true);
+                            }
+                        } else {
+                            logger.warn("Connection lost no ping from YIO DOCK");
+                            YIOREMOTEHANDLESTATUS_actualstatus = YIOREMOTEHANDLESTATUS.CONNECTION_FAILED;
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                    "Connection lost no ping from YIO DOCK");
+                            updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+                            pollingJob.cancel(true);
+                        }
+                    } catch (InterruptedException e) {
+                        logger.warn("Error during initializing the WebSocket polling Thread {}", e.toString());
+                    }
+                }
+            };
+            try {
+                pollingJob = scheduler.scheduleWithFixedDelay(heartbeatpolling, 0, 30, TimeUnit.SECONDS);
+            } catch (Exception e) {
+                updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
+                logger.warn("Error during starting the WebSocket polling Thread {}", e.toString());
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public void dispose() {
+        updateChannelString(GROUP_OUTPUT, YIODOCKSTATUS, "Connection/Configuration Error");
```

Comment:
The framework takes care of the Channel values when the Thing is disposed. This can be removed.

Reply:
done